### PR TITLE
[Service Bus] add test to check halt of ALR on session voluntary close

### PR DESCRIPTION
**Pull request by swathipil**
fixes: #14336

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/tests/test_sessions.py [hunk 1]
```diff
@@ -731,6 +731,27 @@ def lock_lost_callback(renewable, error):
             renewer.close()
             assert len(messages) == 2
 
+        # test voluntary halt of auto lock renewer when session is closed
```
Comment (swathipil) [inline sdk/servicebus/azure-servicebus/tests/test_sessions.py:734]: hey @rakshith91, does this test look similar to what you did when you were [trying to reproduce](https://github.com/Azure/azure-sdk-for-python/issues/14336#issuecomment-721246436)?

Comment (rakshith91) [reply sdk/servicebus/azure-servicebus/tests/test_sessions.py:734]: yes - this seems right to me

**Review by rakshith91 — APPROVED** [review]

File: sdk/servicebus/azure-servicebus/tests/test_sessions.py (hunk 1 above)
Comment (yunhaoling) [reply sdk/servicebus/azure-servicebus/tests/test_sessions.py:734]: looks good to me -- we should cover async well.

**Pull request merged**